solution/index.js

```diff
@@ -0,0 +1,554 @@
+    }
+}
+
+// ===> On right click <===
+function contextMenuTask(event){
+    event.preventDefault();          
+
+    changeSelectedTask(event.target);
+    
+    const { clientX: mouseX, clientY: mouseY } = event;
+    
+    contextMenu.style.top = `${mouseY}px`;
+    contextMenu.style.left = `${mouseX}px`;
+    
+    contextMenu.classList.add("visible");
+
+    document.addEventListener("click", OnContextMenuClick);
+
+}
+
+function OnContextMenuClick(event){    
+    if (event.target.offsetParent == contextMenu){ 
+        if (event.target.dataset.type === "Delete" ){                        
+            if(confirm("Are you sure you want to delete " + SelectedTaskName + " task?")){
+                deleteTask();
+                reloadTasksPage(searchInput.value);  
+            }                  
+        }               
+    }
+    contextMenu.classList.remove("visible");
+    document.removeEventListener("click", OnContextMenuClick);    
+}
+
+function dragEnter(event){
+    event.target.classList.add('droppable-hovered');    
+}
+
+function dragLeave(event){    
+   event.target.classList.remove('droppable-hovered');   
+}
+
+function dragDrop(event){
+    event.target.classList.remove('droppable-hovered');
+    let target = event.target;
+    const taskManagerDataJSON = getLocalStorageTasks(); 
+    const sectionArray = taskManagerDataJSON[SelectedTask.dataset.section];
+    let index = sectionArray.indexOf(SelectedTaskName);
+    // Remove from previous sction
+    sectionArray.splice(index, 1);
+    localStorage.setItem('tasks', JSON.stringify(taskManagerDataJSON));
+    //Add to the new one in a specific index
+    let indexDrop = parseInt(target.dataset.drop_div);    
+    if(target.dataset.section === SelectedTask.dataset.section && indexDrop !== 0 && index < indexDrop){
+        AddToSection(SelectedTask.textContent, target.dataset.section, indexDrop - 1);
+    }
+    else{
+        AddToSection(SelectedTask.textContent, target.dataset.section, target.dataset.drop_div);
+    }
+    reloadTasksPage(searchInput.textContent); 
+    displayError(false);       
+}
+
+function onTaskDragStart(event){
+    //event.preventDefault();
+    changeSelectedTask(event.target);
+    
+    event.target.style.opacity = .9;    
+}
+
+function onTaskDragEnd(event){
+    event.target.style.opacity = "";
+}
+
+function onTaskDragOver(event){
+    event.preventDefault();
+}
+
+function onMouseMove(event){
+    let loader = document.querySelector(".loader");
+    const { clientX: mouseX, clientY: mouseY } = event;
+    
+    loader.style.top = `${mouseY}px`;
+    loader.style.left = `${mouseX}px`;
+    
+}
+
+// ==================================
+// ====== Server Function ============
+// ==================================
+async function SaveTasksOnServer(){
+    const taskManagerDataString = localStorage.getItem('tasks');   
+    
+    loaderDisplay(true); 
+    document.addEventListener('mousemove', onMouseMove);    
```

Comment:
Do not add an event to the document level. it will run every time your mouse move and that is not a good logic.

Reply:
But I added this event listener and right a after fetch I removed the event. So its will not run every time, only after the event is added.